src/dex/nerve/getstate-multicall.ts

```diff
@@ -0,0 +1,124 @@
+import _ from 'lodash';
+import { DeepReadonly } from 'ts-essentials';
+import type { NerveEventMetapool } from './nerve-metapool';
+import {
+  PoolState,
+  MetapoolState,
+  EventPoolOrMetapool,
+  PoolOrMetapoolState,
+} from './types';
+
+async function _getManyPoolStates(
+  pools: EventPoolOrMetapool[],
+  multi: any,
+  blockNumber: number | 'latest' = 'latest',
+): Promise<PoolOrMetapoolState[]> {
+  const calldata = _(pools)
+    .map((pool: EventPoolOrMetapool) => [
+      {
+        target: pool.address,
+        callData: pool.poolIface.encodeFunctionData('swapStorage', []),
+      },
+      {
+        target: pool.lpToken,
+        callData: pool.lpTokenIface.encodeFunctionData('totalSupply', []),
+      },
+      _.range(0, pool.numTokens).map(poolIndex => ({
+        target: pool.address,
+        callData: pool.poolIface.encodeFunctionData('getTokenBalance', [
+          poolIndex,
+        ]),
+      })),
+      (pool as NerveEventMetapool).basePool
+        ? [
+            {
+              target: pool.address,
+              callData: pool.poolIface.encodeFunctionData(
+                'metaSwapStorage',
+                [],
+              ),
+            },
+          ]
+        : null,
+    ])
+    .flattenDeep()
+    .filter(a => !!a)
+    .value();
+
+  const data = await multi.methods.aggregate(calldata).call({}, blockNumber);
+
+  let p = 0;
+  return pools.map((pool: EventPoolOrMetapool) => {
+    const swapStorage = pool.poolIface.decodeFunctionResult(
+      'swapStorage',
+      data.returnData[p++],
+    );
+    const _state: PoolState = {
+      initialA: BigInt('0'),
+      futureA: BigInt('0'),
+      initialATime: BigInt('0'),
+      futureATime: BigInt('0'),
+      swapFee: BigInt('0'),
+      adminFee: BigInt('0'),
+      defaultDepositFee: BigInt('0'),
+      defaultWithdrawFee: BigInt('0'),
+      lpToken_supply: BigInt('0'),
+      balances: [BigInt('0')],
+      tokenPrecisionMultipliers: [BigInt('0')],
+      isValid: true,
+    };
+    return (pool as NerveEventMetapool).basePool
+      ? {
+          ..._state,
+          baseVirtualPrice: BigInt('0'),
+          baseCacheLastUpdated: BigInt('0'),
+        }
+      : _state;
+  });
+}
```

Comment:
This doesn't look correct at all to me! you are fetching so many things but you are only decoding the `swapStorage`!

Reply:
Sorry for misleading you. As we discussed in DMs, this part is not used anymore. Deleted it.